ui/SpinInfoBar.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useEffect, useState } from 'react';
+import { useDark } from './lib/useDark';
+
+export const SpinInfoBar = () => {
+  const [spinStack, setSpinStack] = useState<string[]>([]);
+  const dark = useDark();
+
+  useEffect(() => {
+    console.log('spin: setting up...');
+    let available = false;
+    const spin = new EventSource('/~_~/spin', { withCredentials: true });
+
+    spin.onopen = () => {
+      console.log('spin: opened stream');
+      available = true;
+    };
+
+    spin.onmessage = (e) => {
+      console.log('spin message:', e.data);
+      // The data is expected to be a path like /stack3/stack2/stack1/stack0
+      // We need to parse it into an array of stack names
+      if (e.data && typeof e.data === 'string') {
+        // Remove leading slash if present and split by slash
+        const stackPath = e.data.startsWith('/') ? e.data.substring(1) : e.data;
+        const stackArray = stackPath.split('/').filter(Boolean);
+        setSpinStack(stackArray);
+      }
+    };
+
+    spin.onerror = (e) => {
+      console.error('spin: eventsource error:', e);
+      if (available) {
+        window.setTimeout(() => {
+          if (spin.readyState !== EventSource.CLOSED) {
+            return;
+          }
+          console.log('spin: reconnecting...');
+          // We would call the setup function again here, but we don't want to
+          // create multiple event listeners in this effect
+        }, 10000);
+      }
+    };
```

Comment:
This doesn't actually reconnect, just prints. Probably shouldn't even be conditional on `available`, in case it errors/drops before it completes setup/calls `onopen()`.

Reply:
Yea this doesn't do anything. I removed lines 32-41.